src/data/p5.TypedDict.js

```diff
@@ -0,0 +1,591 @@
+/**
+ * @module Data
+ * @submodule Dictionary
+ * @for p5.TypedDict
+ * @requires core
+ *
+ * This module defines the p5 methods for the p5 Dictionary classes
+ * these classes StringDict and NumberDict are for storing and working
+ * with key, value pairs
+ */
+
+'use strict';
+
+var p5 = require('../core/core');
+
+/**
+ *
+ * Creates a new instance of p5.StringDict using the key, value pair
+ * or object you provide.
+ *
+ * @method createStringDict
+ * @param {String|Object} key or object
+ * @param {String} value
+ * @return {p5.StringDict}
+ *
+ */
+
+p5.prototype.createStringDict = function() {
+  return new p5.StringDict(arguments[0], arguments[1]);
+};
+
+/**
+ *
+ * Creates a new instance of p5.NumberDict using the key, value pair
+ * or object you provide.
+ *
+ * @method createNumberDict
+ * @param {Number|Object} key or object
+ * @param {Number} value
+ * @return {p5.NumberDict}
+ *
+ */
+
+p5.prototype.createNumberDict = function() {
+  return new p5.NumberDict(arguments[0], arguments[1]);
+};
+
+/**
+ *
+ * Base class for all p5.Dictionary types. More specifically
+ * typed Dictionary objects inherit from this
+ *
+ */
+
+p5.TypedDict = function() {
+  this.data = {};
+  if(arguments[0][0] instanceof Object) {
+    this.data = arguments[0][0];
+  } else {
+    this.data[arguments[0][0]] = arguments[0][1];
+  }
+  return this;
+};
+
+/**
+ * Returns the number of key-value pairs currently in Dictionary object
+ *
+ * @method size
+ * @return {Number} the number of key-value pairs in Dictionary object
+ *
+ * @example
+ * <div><code>
+ *
+ * function setup() {
+ *   var myDictionary = createNumberDict(1, 10);
+ *   myDictionary.create(2, 20);
+ *   myDictionary.create(3, 30);
+ *   var amt = myDictionary.size(); // value of amt is 3
+ * }
+ *
+ * </code></div>
+ *
+ */
+p5.TypedDict.prototype.size = function(){
+  return Object.keys(this.data).length;
+};
+
+/**
+ * Returns true if key exists in Dictionary
+ * otherwise returns false
+ *
+ * @method hasKey
+ * @param {Number|String} key that you want to access
+ * @return {Boolean} whether that key exists in Dictionary
+ *
+ * @example
+ * <div><code>
+ *
+ * function setup() {
+ *   var myDictionary = createStringDict('p5', 'js');
+ *   print(myDictionary.hasKey('p5')); // logs true to console
+ * }
+ *
+ * </code></div>
+ *
+ */
+
+p5.TypedDict.prototype.hasKey = function(key) {
+  return this.data.hasOwnProperty(key);
+};
+
+/**
+ * Returns value stored at supplied key.
+ *
+ * @method get
+ * @param {Number|String} key that you want to access
+ * @return {Number|String} the value stored at that key
+ *
+ * @example
+ * <div><code>
```

Comment:
if the examples aren't drawing anything, let's add class="norender" so we don't have blank gray squares in the ref.